lib/oppen/print_stack.rb

```diff
@@ -0,0 +1,201 @@
+# frozen_string_literal: true
+
+# Oppen.
+module Oppen
+  # Class that reprents a stack that builds an output string
+  # using the values of the tokens that were pushed into it.
+  class PrintStack
+    # Class that represents an item in the print stack.
+    class PrintStackEntry
+      # @return [Integer] Indentation level.
+      attr_reader :offset
+      # @return [Token::BreakType] (Called break in the original paper).
+      attr_reader :break_type
+
+      def initialize(offset, break_type)
+        @offset = offset
+        @break_type = break_type
+      end
+    end
+
+    # @return [Integer] Current available space (Called index in the original paper).
+    attr_reader :space
+
+    def initialize(line_width)
+      # Array representing the stack of PrintStackEntries.
+      @items = []
+
+      # Current available space (Called index in the original paper).
+      @space = line_width
+
+      # Page margin (Called length in the original paper).
+      @margin = line_width
+
+      # IO element that builds the output.
+      @output = StringIO.new
+    end
+
+    # Returns the output of the print stack
+    #
+    # @return [StringIO]
+    def output
+      @output.string
+    end
+
+    # Core method responsible for building the print stack and the output string.
+    #
+    # Called Print in the original paper.
+    #
+    # @param token [Token]
+    # @param token_length [Integer]
+    #
+    # @return [Nil]
```

Comment:
Now that It's looking more proper, all the `Called … in the original paper` are worth being put in a `@note`.

And `@note` should always be before all other annotations, after the main description.